### ISSUE TITLE
Incompatible encrypted sessions with update

### ISSUE THREAD

**Issue by nick** [nlf/connect-mysql#34]
It seems the format of the encrypted session has changed since the last update, causing an error to be thrown if when the 'old' JSON formatted session is decrypted.

Since this is a breaking change, can we call this update version 3.0.0 instead of 2.1.3?

**Comment by jas-** [comment]
How old are your encrypted sessions?

The API technically hasn't changed for getting/setting session data.

The structure of the session data was changed to address a security issue with regards to data tampering on decryption.

According to the semantic versioning this would be a big fix or even a minor revision.

Are you using another piece of middleware to read session data that is breaking due to this change?

**Comment by nick** [comment]
The sessions are from just before this change. In order to make this a non-breaking change the session database should be truncated after upgrading, which would cause all logged in users to be logged out.

With a fresh installation and a truncated table, I get the following error:

```
SyntaxError: Unexpected token a in JSON at position 2
    at JSON.parse (<anonymous>)
    at Query._callback (/master/node_modules/connect-mysql/lib/connect-mysql.js:280:34)
    at Query.Sequence.end (/master/node_modules/connect-mysql/node_modules/mysql/lib/protocol/sequences/Sequence.js:88:24)
    at Query._handleFinalResultPacket (/master/node_modules/connect-mysql/node_modules/mysql/lib/protocol/sequences/Query.js:139:8)
    at Query.EofPacket (/master/node_modules/connect-mysql/node_modules/mysql/lib/protocol/sequences/Query.js:123:8)
    at Protocol._parsePacket (/master/node_modules/connect-mysql/node_modules/mysql/lib/protocol/Protocol.js:279:23)
    at Parser.write (/master/node_modules/connect-mysql/node_modules/mysql/lib/protocol/Parser.js:76:12)
    at Protocol.write (/master/node_modules/connect-mysql/node_modules/mysql/lib/protocol/Protocol.js:39:16)
    at Socket.<anonymous> (/master/node_modules/connect-mysql/node_modules/mysql/lib/Connection.js:103:28)
    at emitOne (events.js:115:13)
    at Socket.emit (events.js:210:7)
    at addChunk (_stream_readable.js:252:12)
    at readableAddChunk (_stream_readable.js:239:11)
    at Socket.Readable.push (_stream_readable.js:197:10)
    at TCP.onread (net.js:589:20)
```

Looking through the code, I see no call to `decryptData` - I think that should be happening around [here](https://github.com/nlf/connect-mysql/blob/master/lib/connect-mysql.js#L282-L284)?

**Comment by jas-** [comment]
Thanks for reporting this @nick. `v2.1.4` addresses the issue.

Looks like a test harness should be implemented to prevent oversights like this in the future.

**Comment by nick** [comment]
@jas- I didn't see any changes apart from a version bump in the diff:

https://github.com/nlf/connect-mysql/commit/a58c087e4dc99045a0e5ca883f5da2147cc4f025

**Comment by jas-** [comment]
@nick fixed w/ `v2.1.5`

**Comment by edgardz** [comment]
I'm getting the same error with a fresh instalation and a truncated table. 
Even using v2.1.5

**Comment by nick** [comment]
@edgardz I'd suggest sticking with version 2.1.1 - we were unable to get 2.1.5 working either

**Comment by jas-** [comment]
@nick @edgardz

My sincerest apologies. It seems I was not thorough enough in my testing to have assumed it was resolved with `2.1.5`; this patch release `v2.1.6` was tested much more thoroughly and will address the issues you were both experiencing. Again my apologies.

**Comment by jas-** [comment]
Marking as fixed with `2.1.6`, thanks again for reporting this issue.
